hpedockerplugin/cmd/cmd_deleteshare.py

```diff
@@ -19,154 +22,179 @@ def __init__(self, file_mgr, share_info):
         self._fpg_name = share_info['fpg']
 
     def execute(self):
-        LOG.info("Delting share %s..." % self._share_info['name'])
+        share_name = self._share_info['name']
+        LOG.info("Deleting share %s..." % share_name)
         # Most likely nothing got created at the backend when share is
         # not in AVAILABLE state
-        if self._share_info['status'] != 'AVAILABLE':
-            self._delete_share_from_etcd(self._share_info['name'])
-            return json.dumps({u"Err": ''})
+        if self._share_info['status'] == 'FAILED':
+            LOG.info("Share %s is not in FAILED state. Removing from ETCD..."
```

Comment:
@imran-ansari Needs Sentence correction. It should be "Share %s is in FAILED state."

Reply:
:)... you are right sir... will fix that.